torch/csrc/jit/script/init.cpp

```diff
@@ -157,10 +202,29 @@ struct ModuleValue : public SugaredValue {
   virtual std::vector<Value*> call(SourceRange loc, Method & caller, at::ArrayRef<Value*> inputs, List<Attribute> attributes, size_t n_outputs) override {
     return attr(loc, caller, "forward")->call(loc, caller, inputs, attributes, n_outputs);
   }
+
+  virtual std::vector<std::shared_ptr<SugaredValue>> unrolledFor(SourceRange loc, Method& m) override {
+    py::object py_module = py::cast(module);
+    if(!py::isinstance(py_module, py::module::import("torch.jit").attr("_ConstModuleList")))
+      return SugaredValue::unrolledFor(loc, m);
+    std::vector<std::shared_ptr<SugaredValue>> result;
+    for(py::handle module : py_module) {
+      py::object obj = py::reinterpret_borrow<py::object>(module);
+      if(py::isinstance<Module>(obj)) {
+        auto r = py::cast<std::shared_ptr<Module>>(obj);
+        result.push_back(std::make_shared<ModuleValue>(r));
+      } else {
+        result.push_back(std::make_shared<ConstantPythonValue>(obj));
```

Comment:
How would it be possible to get a constant here? Isn't this supposed to be a Module-only iterator?

Reply:
It can be a non-script module, which we use a PythonValue for.